src/ipips/ipip-0342.md

```diff
@@ -0,0 +1,366 @@
+---
+title: "IPIP-0342: Content Router Ambient Discovery"
+date: 2022-11-11
+ipip: proposal
+editors:
+  - name: Will Scott
+    github: willscott
+    affiliation:
+      name: Protocol Labs
+      url: https://protocol.ai/
+  - name: Masih Derkani
+    github: masih
+    affiliation:
+      name: Protocol Labs
+      url: https://protocol.ai/
+relatedIssues:
+  - https://hackmd.io/bh4-SCWfTBG2vfClG0NUFg
+  - https://github.com/ipfs/kubo/issues/9150
+  - https://github.com/filecoin-project/storetheindex/issues/823
+order: 342
+tags: ['ipips']
+---
+
+## Summary
+
+Design discovery and ranking of public content routers on the network.
+
+## Motivation
+
+The Interplanetary stack has slowly opened itself to support extensibility of
+the content routing subsystem. This extensibility is used today by network
+indexers, like https://cid.contact/, to bridge content from large providers
+that cannot practically provide all content to the IPFS DHT. A missing piece
+of this story is that there is not a process by which IPFS nodes can discover
+these alternative content routing systems automatically. This IPIP proposes
+a mechanism by which IPFS nodes can discover and make use of content routing
+systems.
+
+The mechanism proposed by this IPIP, where nodes gossip preferred routers
+to their connected peers, can also have broader applications. The same
+mechanism could be used for external IPNS, peer routers, relays, or DNS
+resolvers. We point out the label allowing re-use of this mechanism for
+other systems in the [protocol design](#1-content-routing-as-a-libp2p-protocol),
+but otherwise leave the concrete design for other systems to subsequent
+IPIPs.
+
+There is currently not a process by which IPFS nodes can use alternative
+content routing systems automatically. This has led to a reliance on
+centralized systems, like the hydra boosters, to fill the gap and offer
+content only available in network indexers to current IPFS nodes. This strategy
+is also insufficient long term because:
+1. It limits speed to the use of a globally distributed kademlia DHT
+2. It is insufficient for providing content in applications where content grows
+    super-linearly to peers, such that the burden on a traditional DHT would
+    become unsustainable.
+
+## Detailed design
+
+This spec is designed for the ability of IPFS nodes to automatically discover
+and make use of 'content routers'. Content routers are services which are able
+to fulfill IPFS's [ContentRouting](https://github.com/libp2p/go-libp2p/blob/master/core/routing/routing.go#L26)
+API. These routers currently are considered to directly support queries using
+the protocols specified by
+[IPIP-337](https://github.com/ipfs/specs/pulls/337)
+and/or
+[IPIP-327](https://github.com/ipfs/specs/pull/327).
```

Comment:
Would it be worth calling out these names as well, so users unfamiliar with the ipip-### dont have to click through